test/test-users.js

```diff
@@ -0,0 +1,402 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const spies = require('chai-spies');
+const mongoose = require('mongoose');
+const app = require('../server').app;
+
+const {
+  User
+} = require('../users/models');
+const should = chai.should();
+
+var UrlPattern = require('url-pattern');
+const pattern = {
+  "/users": new UrlPattern('/users'),
+  "/users/:userId": new UrlPattern('/users/:userId')
+}
+
+chai.use(chaiHttp);
+chai.use(spies);
+
+describe('Test /user endpoints with empty database', () => {
+  beforeEach((done) => {
+    // Approach #1 Drop database and `ensureIndexes()`
+    mongoose.connection.db.dropDatabase()
```

Comment:
Yeah, so if I understand the `ensureIndex` thing right, then I think we'd want to do:

```js
beforeEach(function() {
  // ensure indexes
  // seed database with test data
})
```

Or maybe it makes sense to move the ensureIndex part to the `before`?  Sounds like `ensureIndex` runs when the server starts, and that only happens once, in the before routine. Would this make sense?

```js
before(function() {
    return runServer(TEST_DATABASE_URL).then(() => { // do ensure index});
  });

  beforeEach(function() {
    return promiseThatSeedsDb();
  });
```